blockauth/apple/notification_service.py

```
@@ -63,50 +137,64 @@ def dispatch(self, payload_jwt: str) -> AppleNotificationDispatchResult:
-            except Exception as exc:  # never let an integrator hook bring down the webhook
-                logger.error("apple.notification.trigger_failed", extra={"error_class": exc.__class__.__name__})
-
-        return AppleNotificationDispatchResult(event_type=event_type, handled=handled)
+                    trigger_instance = trigger()
+                    trigger_method = getattr(trigger_instance, "trigger", None)
+                    if callable(trigger_method) and not getattr(trigger_method, "__isabstractmethod__", False):
+                        trigger_method(payload)
+                    else:
+                        run_method = getattr(trigger_instance, "run")
+                        run_method(payload)
+                except Exception as exc:  # never let an integrator hook bring down the webhook
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)